app/assets/javascripts/components/features/compose/components/compose_form.jsx

```diff
@@ -198,7 +198,7 @@ const ComposeForm = React.createClass({
 
           <div style={{ display: 'flex' }}>
             <div style={{ paddingTop: '10px', marginRight: '16px', lineHeight: '36px' }}><CharacterCounter max={500} text={[this.props.spoiler_text, this.props.text].join('')} /></div>
-            <div style={{ paddingTop: '10px' }}><Button text={publishText} onClick={this.handleSubmit} disabled={disabled} /></div>
+            <div style={{ paddingTop: '10px' }}><Button text={publishText} onClick={this.handleSubmit} disabled={disabled || this.props.text.length > 500} /></div>
```

Comment:
Try this `this.props.text.replace(/[\uD800-\uDBFF][\uDC00-\uDFFF]/g, "_").length` instead - otherwise there's wrong counts for emoji and other multibyte characters.